www/js/app.js

```diff
@@ -640,15 +640,63 @@ function focusPrefixOnHomeKey (event) {
+ * @param {archive} archive the archive that needs verification
  * */
 async function verifyLoadedArchive (archive) {
-    const response = await uiUtil.systemAlert(translateUI.t('dialog-sourceverification-alert') ||
-    'Is this ZIM archive from a trusted source?\n If not, you can still read the ZIM file in Safe Mode. Closing this window also opens the file in Safe Mode. This option can be disabled in Expert Settings',
-    translateUI.t('dialog-sourceverification-title') || 'Security alert!', true, translateUI.t('dialog-sourceverification-safe-mode-button') || 'Open in Safe Mode',
-    translateUI.t('dialog-sourceverification-trust-button') || 'Trust Source');
+    // We construct a HTML element to show the user in the alert with the metadata contained in it
+    const metadataLabels = {
+        name: translateUI.t('dialog-metadata-name') || 'Name: ',
+        creator: translateUI.t('dialog-metadata-creator') || 'Creator: ',
+        publisher: translateUI.t('dialog-metadata-publisher') || 'Publisher: ',
+        scraper: translateUI.t('dialog-metadata-scraper') || 'Scraper: '
+    }
+
+    const verificationBody = document.createElement('div');
+
+    // text & metadata box
+    const verificationText = translateUI.t('dialog-sourceverification-alert') || 'Is this ZIM archive from a trusted source?\n If not, you can still read the ZIM file in Safe Mode. Closing this window also opens the file in Safe Mode. This option can be disabled in Expert Settings';
+
+    const metadataBox = document.createElement('div');
+    metadataBox.id = 'modal-archive-metadata-container';
+
+    const verifyName = document.createElement('p');
+    verifyName.id = 'confirm-archive-name';
+    verifyName.classList.add('archive-metadata');
+    verifyName.innerText = metadataLabels.name + (archive.name || '-');
+
+    const verifyCreator = document.createElement('p');
+    verifyCreator.id = 'confirm-archive-creator';
+    verifyCreator.classList.add('archive-metadata')
+    verifyCreator.innerText = metadataLabels.creator + (archive.creator || '-');
+
+    const verifyPublisher = document.createElement('p');
+    verifyPublisher.id = 'confirm-archive-publisher';
+    verifyPublisher.classList.add('archive-metadata');
+    verifyPublisher.innerText = metadataLabels.publisher + (archive.publisher || '-');
+
+    const verifyScraper = document.createElement('p');
+    verifyScraper.id = 'confirm-archive-scraper';
+    verifyScraper.classList.add('archive-metadata');
+    verifyScraper.innerText = metadataLabels.scraper + (archive.scraper || '-');
+
+    const verifyWarning = document.createElement('p');
+    verifyWarning.id = 'modal-archive-metadata-warning';
+    verifyWarning.innerText = translateUI.t('dialog-metadata-warning') || 'Warning: above data can be spoofed!';
+
+    metadataBox.append(verifyName, verifyCreator, verifyPublisher, verifyScraper);
+    verificationBody.append(verificationText, metadataBox, verifyWarning);
```

Comment:
I think it should be OK to use `append()` here because the code will not be executed by IE11 or by any browser that does not support Service Workers.